core/src/main/java/org/apache/druid/metadata/DynamicConfigProvider.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.metadata;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.apache.druid.guice.annotations.ExtensionPoint;
+
+import java.util.Map;
+
+/**
+ * This is used to get [secure] configuration in various places in an extensible way.
+ */
+@ExtensionPoint
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = MapBasedDynamicConfigProvider.class)
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "map", value = MapBasedDynamicConfigProvider.class),
+})
+public interface DynamicConfigProvider
+{
+  Map<String, String> getConfig();
```

Comment:
should this return `Map<String, Object>` to support a wider range of possible configuration value types since any extension other than Kafka could use it?

Reply:
Intention of this interface certainly is to be used everywhere we get extensible user config (currently the places that are using `PasswordProvider`), while a cursory look says that `Map<String, String>` would be sufficient for all the places but I see the desire to have it be more generic. I am probably gonna parameterize the interface to `DynamicConfigProvider<T>` so that client code would still be able to work without typecasting but interface would still serve use case of any value types.